webviz_subsurface_components/py_expression_eval.py

```diff
@@ -0,0 +1,483 @@
+
+You are free to use and modify this code in anyway you find useful. Please leave this comment
+in the code to acknowledge its original source. If you feel like it, I enjoy hearing about
+projects that use my code, but don't feel like you have to let me know or ask permission.
+"""
+# pylint: skip-file
+
+from __future__ import division
+
+import re
+
+import numpy as np
+
+TNUMBER = 0
+TOP1 = 1
+TOP2 = 2
+TVAR = 3
+TFUNCALL = 4
+
+
+class Token:
+    def __init__(self, type_, index_, prio_, number_):
+        self.type_ = type_
+        self.index_ = index_ or 0
+        self.prio_ = prio_ or 0
+        self.number_ = number_ if number_ != None else 0
+
+    def toString(self):
+        if self.type_ == TNUMBER:
+            return self.number_
+        if self.type_ == TOP1 or self.type_ == TOP2 or self.type_ == TVAR:
+            return self.index_
+        elif self.type_ == TFUNCALL:
+            return "CALL"
+        else:
+            return "Invalid Token"
+
+
+class Expression:
+    """
+    Expression based on Expression in py_expression_eval, modifications are done.
+
+    `Adjustments:`
+    Removed unused functionality as simplify, substitute and toString conversion.
+
+    Removed possibility to assign functions to variables during evaluate(). Thereby variables
+    function is simplified as well.
+
+    `Removed functions:`
+
+    - simplify()
+    - substitute()
+    - toString()
+    - symbols()
+
+    `Adjusted functions:`
+
+    - variables() - As functions does not exist, previous symbols function is equal variables
+    """
+
+    def __init__(self, tokens, ops1, ops2):
+        self.tokens = tokens
+        self.ops1 = ops1
+        self.ops2 = ops2
+
+    def evaluate(self, values):
+        values = values or {}
+        nstack = []
+        L = len(self.tokens)
+        for item in self.tokens:
+            type_ = item.type_
+            if type_ == TNUMBER:
+                nstack.append(item.number_)
+            elif type_ == TOP2:
+                n2 = nstack.pop()
+                n1 = nstack.pop()
+                f = self.ops2[item.index_]
+                nstack.append(f(n1, n2))
+            elif type_ == TVAR:
+                if item.index_ in values:
+                    nstack.append(values[item.index_])
+                else:
+                    raise Exception("undefined variable: " + item.index_)
+            elif type_ == TOP1:
+                n1 = nstack.pop()
+                f = self.ops1[item.index_]
+                nstack.append(f(n1))
+            elif type_ == TFUNCALL:
+                n1 = nstack.pop()
+                f = nstack.pop()
+                if callable(f):
+                    if type(n1) is list:
+                        nstack.append(f(*n1))
+                    else:
+                        nstack.append(f(n1))
+                else:
+                    raise Exception(f + " is not a function")
```

Comment:
We should improve the code here, since we have the opportunity now. I.e. instead of raising a general `Exception` maybe instead raise a `ValueError` exception, or a custom error like `ParserError`:

```python

class ParserError(Exception):
    pass


def some_function():
    raise ParserError(f"{f} is not a function")
```
This also enables us to remove the not-so-nice `pylint: disable=broad-exception` in other parts of the code here (as well as in the downstream package `webviz-subsurface` potentially).